File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
	const dictionary = [];
	const vowels = ["A", "E", "I", "O", "U"];
    
	function generateWords(currentWord) {
		if (currentWord.length === 5) {
			return;
		}
        
		for (let i = 0; i < vowels.length; i++) {
			const newWord = currentWord + vowels[i];
			dictionary.push(newWord);
			generateWords(newWord);
		}
	}
	generateWords("");
	return dictionary.indexOf(word) + 1;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    dictionary = []
    vowels = ["A", "E", "I", "O", "U"]
    
    def generate_words(current_word):
        if len(current_word) == 5:
            return
        
        for vowel in vowels:
            new_word = current_word + vowel
            dictionary.append(new_word)
            generate_words(new_word)
            
    generate_words("")
    return dictionary.index(word) + 1